fee-market/src/subtract_fee.rs

```diff
@@ -123,14 +124,17 @@ pub trait SubtractFeeModule:
         }
 
         let final_payment = self.subtract_fee_by_type(payment, total_transfers, opt_gas_limit);
-        let _ = self
-            .tokens_for_fees()
+
+        self.tokens_for_fees()
             .insert(final_payment.fee.token_identifier.clone());
+
         self.accumulated_fees(&final_payment.fee.token_identifier)
             .update(|amt| *amt += &final_payment.fee.amount);
 
-        self.send()
-            .direct_non_zero_esdt_payment(&original_caller, &final_payment.remaining_tokens);
+        self.tx()
```

Comment:
You need a check if payment is non zero so you won't make transactions with 0 value. Or is the transfer function already checking this?

Reply:
added `require`